fairing/builders/config_maps.py

```diff
@@ -0,0 +1,61 @@
+import os
+import subprocess
+import tempfile
+import shutil
+
+from kubernetes import client
+
+from fairing import notebook_helper
+from .builder import BuilderInterface
+from fairing.backend import kubernetes
+
+
+class ConfigMapBuilder(BuilderInterface):
+
+    def __init__(self, notebook_name=None):
+        if notebook_name is None:
+            self.notebook_name = notebook_helper.get_notebook_name()
+        else:
+            self.notebook_name = notebook_name
+
+    def execute(self, namespace, job_id, base_image):
+        nb_full_path = os.path.join(os.getcwd(), self.notebook_name)
+        temp_dir = tempfile.mkdtemp()
+        code_path = os.path.join(temp_dir, 'code.py')
+        try:
+            cmd = "jupyter nbconvert --to python {nb_path} --output {output}"
+            .format(
+                nb_path=nb_full_path,
+                output=code_path
+            ).split()
+
+            subprocess.check_call(cmd)
+            with open(code_path, 'rb') as f:
+                code = f.read()
+        finally:
+            shutil.rmtree(temp_dir, ignore_errors=True)
+
+        client.V1ConfigMap(
+            api_version="v1",
+            data={"code.py": code},
+            metadata=client.V1ObjectMeta(name=self.job_id)
+        )
+        k8s = kubernetes.KubeManager()
+        k8s.create_config_map(namespace, config_map)
+        return self._generate_pod_spec(job_id, base_image)
+
+    def _generate_pod_spec(self, job_id, base_image):
```

Comment:
Not sure if it would be this PR but would be great to provide this pod-spec using a configmap(that way specifying images, environments, and other mounts is completely upto the user).